romancal/associations/skycell_asn.py

```diff
@@ -0,0 +1,172 @@
+"""Create an association based on skycells"""
+
+import argparse
+import json
+import logging
+import sys
+
+import numpy as np
+import roman_datamodels as rdm
+
+import romancal.patch_match.patch_match as pm
+from romancal.associations import asn_from_list
+
+__all__ = ["skycell_asn"]
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+logger.setLevel("INFO")
+
+
+def skycell_asn(self):
+    """Create the associaton from the list"""
+    all_patches = []
+    file_list = []
+    for file_name in self.parsed.filelist:
+        cal_file = rdm.open(file_name)
+        filter_id = cal_file.meta.instrument.optical_element
+        file_patch_list = pm.find_patch_matches(cal_file.meta.wcs)
+        logger.info(f"Patch List:   {file_name}  {file_patch_list[0]}")
+        file_list.append([file_name, file_patch_list[0]])
+        all_patches.append(file_patch_list[0])
+
+    unique_patches = np.unique(np.concatenate(all_patches))
+    for item in unique_patches:
+        member_list = []
+        patch_name = pm.PATCH_TABLE[item]["name"]
+        for a in file_list:
+            if np.isin(item, a[1]):
+                member_list.append(a[0])
+
+        # grab all the wcs parameters needed for generate_tan_wcs
+        projcell_info = dict(
+            [
+                ("pixel_scale", float(pm.PATCH_TABLE[item]["pixel_scale"])),
+                ("ra_cent", float(pm.PATCH_TABLE[item]["ra_projection_center"])),
+                ("dec_cent", float(pm.PATCH_TABLE[item]["dec_projection_center"])),
+                ("shiftx", float(pm.PATCH_TABLE[item]["x0_projection"])),
+                ("shifty", float(pm.PATCH_TABLE[item]["y0_projection"])),
+                ("nx", int(pm.PATCH_TABLE[item]["nx"])),
+                ("ny", int(pm.PATCH_TABLE[item]["ny"])),
+            ]
+        )
+        program_id = member_list[0][1:6]
+        root_asn_name = self.parsed.output_file_root
+        product_type = self.parsed.product_type
+        product_release = self.parsed.release_product
+        suffix = "i2d"
+        sep = "_"
+        asn_file_name = (
+            root_asn_name
+            + sep
+            + patch_name
+            + sep
+            + product_type
+            + sep
+            + filter_id
+            + sep
+            + product_release
+            + sep
+            + suffix
+        )
+        with open(asn_file_name + "_asn.json", "w") as outfile:
+            prompt_product_asn = asn_from_list.asn_from_list(
+                member_list, product_name=asn_file_name
+            )
+            prompt_product_asn["asn_type"] = "image"
+            prompt_product_asn["program"] = program_id
+            prompt_product_asn["target"] = patch_name
+            prompt_product_asn["skycell_wcs_info"] = json.dumps(projcell_info)
```

Comment:
I think I would have imagined leaving this as a dictionary, rather than converting it to a string via dumps?  What's the motivation for stringifying this?

Reply:
Keeping this a string  allows the other unit tests to pass since we're using one json schema file at the moment. Since a dictionary is more convenient we should add schema for the skycell asn - at a later date.